File: server/src/game/tools/conf/output/ts/SignConfig.ts
class SignConfig {
    private static _dic:Dictionary<SignConfig> = new Dictionary<SignConfig>();
    private static _arr:Array<SignConfig> = new Array<SignConfig>();

	 public Id :number ;
     public Name :string ;
     public Icon :string ;
     public Bg :string ;
     public Reward : Array<number>  =new Array<number>();
     public Nums : Array<number>  =new Array<number>();
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<SignConfig>{
        return  SignConfig._dic;
    }
    public static get Arr():Array<SignConfig>{
        return SignConfig._arr;
    }
    public static parse(json:any[]){

        for(var i = 0; i < json.length; i++){
            let config:SignConfig = <SignConfig>json[i];
            SignConfig._arr.push(config);
            SignConfig._dic.add(config.Id,config);
        }
    }
}

File: tools/conf/output/go/SignConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type SignConfigElem struct {
	Id int
	Name string
	Icon string
	Bg string
	Reward []int
	Nums []int
	
}

type SignConfig0 struct {
	Dic map[int]*SignConfigElem
	Arr []*SignConfigElem
}

var (
	SignConfig SignConfig0
)

func loadSignConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	SignConfig.Dic = make(map[int]*SignConfigElem)
	SignConfig.Arr = make([]*SignConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &SignConfig.Arr)
	for _,conf := range SignConfig.Arr{
		SignConfig.Dic[conf.Id] = conf
	}
	
}
